src/components/organisms/UserProfile/UpdatePersonalInfo.tsx

```diff
@@ -0,0 +1,202 @@
+import Icons from "@/assets/icons";
+import Avatar from "@/components/atoms/Avatar";
+import { cn } from "@/utils";
+import onBoardingService from "@/utils/apis/onboarding";
+import Alert from "@/utils/base/alerts";
+import { onFormError } from "@/utils/functions/react-hook-form";
+import { PersonalInfoSchema } from "@/utils/schema/details";
+import { type InferSchema } from "@/utils/schema/helpers";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Title, Text, Input, Button, PhoneInput } from "@the_human_cipher/components-library";
+import Image from "next/image";
+import { Dispatch, SetStateAction, useState } from "react";
+import { useForm, SubmitHandler, Controller } from "react-hook-form";
+
+type Inputs = InferSchema<typeof PersonalInfoSchema>;
+
+type InputName = keyof Inputs;
+
+interface Fields {
+  name: InputName;
+  label: string;
+  placeholder?: string;
+  type?: string;
+}
+
+const fields: Fields[] = [
+  {
+    name: "first_name",
+    label: "First Name",
+  },
+  {
+    name: "last_name",
+    label: "Last Name",
+  },
+  {
+    name: "phone_number",
+    label: "Phone Number",
+  },
+  {
+    name: "bio",
+    label: "Bio",
+  },
+];
+
+const UpdatePersonalInfo = ({ isEditMode, setIsEditMode, userProfile }: ProfileProps) => {
+  const { register, formState, handleSubmit, control, watch } = useForm<Inputs>({
+    resolver: zodResolver(PersonalInfoSchema),
+    mode: "onChange",
+    values: userProfile,
+  });
+
+  const selectedFile = watch("avatar");
+
+  const onFormSubmit: SubmitHandler<Inputs> = async (data) => {
+    try {
+      const res = await onBoardingService.updateUserPersonalInfo(data);
+      if (res) {
+        Alert.success("Personal information updated successfully");
+      }
+    } catch (error) {
+      Alert.error(error);
+    }
+  };
+
+  const handleEditClick = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    event.preventDefault();
+    setIsEditMode(false);
+  };
+
+  const getFormError = (key: keyof Inputs) => {
```

Comment:
for subsequent code, i've made this into a function
```tsx
/**
 * @example```tsx
 * const { unwrapFormErrors, assertFormError } = getFormStateError(formState)
 * const xyzError = unWrapErrors("xyz")
 * ```
 */
export const getFormStateError = <Inputs extends FieldValues>(
  formState: FormState<Inputs>
) => {
  const unwrapFormError = (key: keyof Inputs) => {
    const err = formState.errors[key]?.message;
    return err ? String(err) : undefined;
  };

  const assertFormError = (key: keyof Inputs): boolean => {
    return Boolean(formState.errors[key]?.message);
  };

  return { unwrapFormError, assertFormError };
};
```

Reply:
this doesnt require you to make any changes, but we might have to refactor later